### PR TITLE
Deprecate legacy classes and interfaces

### PR DESCRIPTION
**Pull request by peendebak**
* Deprecate instrument adapters
* Deprecate measurement interfaces
* Deprecate post-processing code using qilib dataset

### CODE REVIEW

File: src/tests/unittests/measurements/processing/test_process_sawtooth_2d.py [hunk 1]
```diff
@@ -62,8 +63,8 @@ def __dummy_scope_data(time_data, sawteeth_count, period, width):
 
     @staticmethod
     def __dummy_data_array(set_array: DataArray, scope_data: np.ndarray, channel_index: int = 1, trace_number: int = 1):
-        idenifier = 'ScopeTrace_{:03d}'.format(trace_number)
-        label = 'Channel_{}'.format(channel_index)
+        idenifier = f'ScopeTrace_{trace_number:03d}'
```
Comment (QFer) [inline src/tests/unittests/measurements/processing/test_process_sawtooth_2d.py:66]: Idenifier->Identifier

**Review by QFer — COMMENTED** [review]
Why are all these instrument adapters and post_processing deprecated, what do we use instead?

**Comment by peendebak** [comment]
> Why are all these instrument adapters and post_processing deprecated, what do we use instead?

The adapters and post processing have not been used. We have always used the post processing in `qtt.measurements.scans`.

File: src/tests/unittests/measurements/processing/test_process_sawtooth_2d.py (hunk 1 above)
Comment (peendebak) [reply src/tests/unittests/measurements/processing/test_process_sawtooth_2d.py:66]: The spelling mistake was already in the original code and this is part of the deprecated class.

**Review by QFer — APPROVED** [review]

**Pull request merged**